src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -377,29 +377,8 @@ export class VisualizeEmbeddable extends Embeddable<VisualizeInput, VisualizeOut
   };
 
   public supportedTriggers() {
-    // TODO: Report a correct list of triggers for each vis_type.
-    switch (this.vis.type.name) {
-      case 'area':
-      case 'heatmap':
-      case 'histogram':
-      case 'horizontal_bar':
-      case 'line':
-      case 'pie':
-      case 'table':
-      case 'tagcloud':
-        return [VIS_EVENT_TO_TRIGGER.filter];
-      case 'gauge':
-      case 'goal':
-      case 'input_control_vis':
-      case 'markdown':
-      case 'metric':
-      case 'metrics':
-      case 'region_map':
-      case 'tile_map':
-      case 'timelion':
-      case 'vega':
-      default:
-        return [];
-    }
+    return typeof this.vis.type.getSupportedTriggers === 'function'
```

Comment:
:art: Since we use TypeScript here, and (assuming for now) it's safe, we could maybe not check on the function type here, but instead simplify to `return this.vis.type.getSupportedTriggers?.() ?? []`?

Reply:
Yes we can, you are right! Thanx for that :)